src/feature/hs/hs_common.c

```diff
@@ -1843,3 +1843,40 @@ hs_inc_rdv_stream_counter(origin_circuit_t *circ)
     tor_assert_nonfatal_unreached();
   }
 }
+
+/* Return a newly allocated link specifier object that is a copy of dst. */
+link_specifier_t *
+link_specifier_dup(const link_specifier_t *src)
+{
+  link_specifier_t *dup = NULL;
+  uint8_t *buf = NULL;
+
+  if (BUG(!src)) {
+    goto err;
+  }
+
+  ssize_t est_encoded_len = link_specifier_encoded_len(src);
```

Comment:
Is it possible you meant `dst_encoded_len`? I have no idea what `est_` would mean here hehe?

Reply:
"estimated", but that's wrong, it should be the actual length. (Or at least an upper bound on the length, depending on how TRUNNEL_DYNARRAY_LEN() works.)

Reply:
Fixed in 4cbb752f73.